components/tidb_query_vec_executors/src/index_scan_executor.rs

```diff
@@ -104,6 +106,45 @@ impl<S: Storage> BatchIndexScanExecutor<S> {
         })?;
         Ok(Self(wrapper))
     }
+
+    pub fn new_for_analyze(
+        storage: S,
+        config: Arc<EvalConfig>,
+        cols_len: usize,
+        key_ranges: Vec<KeyRange>,
+        is_common_handle: bool,
+    ) -> Result<Self> {
+        let (decode_handle_strategy, handle_column_cnt) = if is_common_handle {
+            (DecodeCommonHandleRowKey, cols_len)
```

Comment:
why `handle_column_cnt  == cols_len`?

Reply:
tidb analyze row key columns the same way as the primary key index.